smt/utils/misc.py

```diff
@@ -8,6 +8,61 @@
 from bisect import bisect_left
 
 
+def standardization(X, y, xlimits=None):
+
+    """
+
+    We substract the mean from each variable. Then, we divide the values of each
+    variable by its standard deviation. If scale_X_to_unit, we scale the input
+    space X to the unit hypercube [0,1]^dim with dim the input dimension.
+
+    Parameters
+    ----------
+
+    X: np.ndarray [n_obs, dim]
+            - The input variables.
+
+    y: np.ndarray [n_obs, 1]
+            - The output variable.
+
+    Returns
+    -------
+
+    X: np.ndarray [n_obs, dim]
+          The standardized input matrix.
+
+    y: np.ndarray [n_obs, 1]
+          The standardized output vector.
+
+    X_offset: list(dim)
+            The mean (or the min if scale_X_to_unit=True) of each input variable.
+
+    y_mean: list(1)
+            The mean of the output variable.
+
+    X_scale:  list(dim)
+            The standard deviation (or the difference between the max and the
+            min if scale_X_to_unit=True) of each input variable.
+
+    y_std:  list(1)
+            The standard deviation of the output variable.
+
+    """
+
+    X_offset = np.mean(X, axis=0)
+    X_scale = X.std(axis=0, ddof=1)
+
+    X_scale[X_scale == 0.0] = 1.0
```

Comment:
Should be better to test < 1e-15 or 100.*sys.float_info.epsilon instead of == 0.